public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -147,7 +152,7 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
   /**
    * The save model which the scene was originally created from
    */
-  private _initialSaveModel?: Dashboard;
+  private _initialSaveModel?: Dashboard | DashboardV2Spec;
```

Comment:
Third part of the fundamental changes: initial save model now would accept two different schemas. For v2 we gonna need the full k8s resource as part of the information will be provided not through the spec, but annotations (i.e. folder id)